tests/bluetooth/tester/src/btp_bap_stream.h

```diff
@@ -0,0 +1,216 @@
+#define MAX_SNK_STREAMS_COUNT MAX(CLIENT_SNK_COUNT, SERVER_SNK_COUNT)
+#define MAX_SRC_STREAMS_COUNT MAX(CLIENT_SRC_COUNT, SERVER_SRC_COUNT)
+#define MAX_STREAMS_COUNT MAX_SNK_STREAMS_COUNT + MAX_SRC_STREAMS_COUNT
+#define MAX_END_POINTS_COUNT CLIENT_SNK_COUNT + CLIENT_SRC_COUNT
+
+struct unicast_group {
+	struct bt_audio_codec_qos qos[CONFIG_BT_BAP_UNICAST_CLIENT_GROUP_STREAM_COUNT];
+	struct bt_bap_unicast_group *cig;
+	bool in_use;
+};
+
+struct audio_stream {
+	struct bt_cap_stream cap_stream;
+	atomic_t seq_num;
+	uint16_t last_req_seq_num;
+	uint16_t last_sent_seq_num;
+	struct k_work_delayable audio_clock_work;
+	struct k_work_delayable audio_send_work;
+};
+
+struct unicast_stream {
+	struct audio_stream audio_stream;
+	uint8_t ase_id;
+	uint8_t conn_id;
+	uint8_t cig_id;
+	uint8_t cis_id;
+	struct bt_audio_codec_cfg codec_cfg;
+	bool already_sent;
+	bool in_use;
+};
+
+struct unicast_connection {
+	bt_addr_le_t address;
+	struct unicast_stream streams[MAX_STREAMS_COUNT];
+	size_t configured_sink_stream_count;
+	size_t configured_source_stream_count;
+	struct bt_audio_codec_qos qos;
+	struct bt_bap_ep *end_points[MAX_END_POINTS_COUNT];
+	size_t end_points_count;
+};
+
+struct broadcast_stream {
+	struct audio_stream audio_stream;
+	struct bt_audio_codec_cfg codec_cfg;
+	uint8_t bis_id;
+	uint8_t subgroup_id;
+	bool bis_synced;
+	uint8_t source_id;
+	bool in_use;
+	bool already_sent;
+};
+
+/* According to BT spec, a Broadcast Source can configure and establish one or more BIGs,
+ * each containing one or more BISes that are used to transport broadcast Audio Streams.
+ * For each BIG, the Broadcast Source shall generate a Broadcast_ID.
+ * For the time being, let's treat broadcast source as one BIG.
+ */
+struct remote_broadcast_source {
+	bt_addr_le_t address;
+	uint32_t broadcast_id;
+	struct broadcast_stream streams[MAX_STREAMS_COUNT];
+	struct bt_bap_stream *sink_streams[MAX_STREAMS_COUNT];
+	struct bt_bap_broadcast_sink *sink;
+	struct bt_audio_codec_qos qos;
+	/* BIS Index bitfield read from Base */
+	uint32_t bis_index_bitfield;
+	/* BIS Index bitfield read from sync request */
+	uint32_t requested_bis_sync;
+	bool assistant_request;
+	uint8_t sink_broadcast_code[BT_AUDIO_BROADCAST_CODE_SIZE];
+	const struct bt_bap_scan_delegator_recv_state *sink_recv_state;
+};
+
+struct local_broadcast_source {
+	uint32_t broadcast_id;
+	struct bt_audio_codec_qos qos;
+	struct broadcast_stream streams[MAX_STREAMS_COUNT];
+	struct bt_audio_codec_cfg subgroup_codec_cfg[CONFIG_BT_BAP_BROADCAST_SRC_SUBGROUP_COUNT];
+	/* Only for BTP BAP commands */
+	struct bt_bap_broadcast_source *bap_broadcast;
+	/* Only for BTP CAP commands */
+	struct bt_cap_broadcast_source *cap_broadcast;
+};
+
+inline struct unicast_stream *stream_bap_to_unicast(struct bt_bap_stream *stream)
+{
+	return CONTAINER_OF(CONTAINER_OF(CONTAINER_OF(stream, struct bt_cap_stream, bap_stream),
+		struct audio_stream, cap_stream), struct unicast_stream, audio_stream);
+}
```

Comment:
What happens if this is accidentally used for a broadcast stream?

Reply:
Then it is a bug :) I'll try to come up with some better names

Reply:
Is there any way we can check if the `bt_bap_stream` is unicast? (hint: `stream->conn` may be an indicator of this)